extensions/wikia/Email/Controller/WeeklyDigestController.class.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Email\Controller;
+
+use \Email\EmailController;
+
+class WeeklyDigestController extends EmailController {
+
+	const LAYOUT_CSS = 'weeklyDigest.css';
+
+	protected $digestData;
+
+	protected function getSubject() {
+		return $this->getMessage( 'emailext-weeklydigest-subject' )->text();
+	}
+
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-weeklydigest-summary' )->text();
+	}
+
+	public function initEmail() {
+		$this->digestData = $this->request->getVal( 'digestData' );
```

Comment:
Yea, I don't have any super clever ideas either. I think the JSON string approach is the best way for now. I'm also going to include a default value for that field so user's of Special:SendEmail can just send it off and see the digest without having to figure out how to insert json with the correct values into that field.